client/src/components/Header/components/language-list.tsx

```diff
@@ -0,0 +1,167 @@
+);
+
+const mapDispatchToProps = {
+  navigate
+};
+
+interface LanguageListProps {
+  t: TFunction;
+  navigate?: (location: string) => void;
+}
+
+export const LanguageList = ({
+  t,
+  navigate
+}: LanguageListProps): JSX.Element => {
+  const [showList, setShowList] = useState(false);
+  const listButtonRef = useRef<HTMLButtonElement>(null);
+
+  const handleClick = (): void => {
+    if (showList) {
+      setShowList(false);
+      return;
+    }
+    setShowList(true);
+  };
+
+  const handleLanguageChange = (
+    event: React.MouseEvent<HTMLButtonElement>
+  ): void => {
+    const selectedLanguage = event.currentTarget.dataset.value;
+    const isSelecetedCurrentLanguage =
+      selectedLanguage === clientLocale || selectedLanguage === undefined;
+
+    event.preventDefault();
+    listButtonRef.current?.focus();
+    setShowList(false);
+    if (isSelecetedCurrentLanguage) {
+      return;
+    }
+    const path = createLanguageRedirect({
+      clientLocale,
+      lang: selectedLanguage
+    });
+    if (navigate) {
+      return navigate(path);
+    }
+  };
+
+  const handleMenuKeyDown = (event: React.KeyboardEvent<HTMLButtonElement>) => {
+    if (event.key === 'Escape') {
+      listButtonRef.current?.focus();
+      setShowList(false);
+      event.preventDefault();
+    }
+  };
+
+  const handleBlur = (event: React.FocusEvent<HTMLButtonElement>) => {
+    if (
+      event.relatedTarget &&
+      !event.relatedTarget.closest('.nav-list') &&
+      event.relatedTarget !== listButtonRef.current
+    ) {
+      setShowList(false);
+    }
+  };
+
+  const handleLastLangaugeKeys = (
+    event: React.KeyboardEvent<HTMLButtonElement>
+  ) => {
+    const DoKeyPress = new Map<string, { select: () => void }>([
+      [
+        'Escape',
+        {
+          select: () => {
+            listButtonRef.current?.focus();
+            setShowList(false);
+            event.preventDefault();
+          }
+        }
+      ],
+      [
+        'Tab',
+        {
+          select: () => {
+            const camperPressedTheShiftKey = event.shiftKey;
+            if (!camperPressedTheShiftKey) {
+              setShowList(false);
+            }
+          }
+        }
+      ]
+    ]);
+    DoKeyPress.get(event.key)?.select();
+  };
+
+  const getHandleLanguageKeys = (
+    event: React.KeyboardEvent<HTMLButtonElement>,
+    languagePosition: number
+  ) => {
+    const lastLanguage = locales.length - 1;
+    if (languagePosition === lastLanguage) {
+      return handleLastLangaugeKeys(event);
+    } else handleMenuKeyDown(event);
+  };
+  return (
+    <>
+      <button
+        id='toggle-lang-button'
+        className='lang-button-nav'
+        title={t('buttons.change-language')}
+        aria-label={t('buttons.change-language')}
+        aria-controls='nav-lang-list'
+        aria-expanded={showList}
+        ref={listButtonRef}
+        onBlur={handleBlur}
+        onClick={handleClick}
+        onKeyDown={handleMenuKeyDown}
```

Comment:
```suggestion
```

I don't think we need to enable the `Esc` key when focus is on the button. The menu can already be toggled with either the `Enter` key or Spacebar. The `Esc` key is not an expected feature for a button and the main Menu button does not have it, so I think we should keep them the same.